src/App.tsx

```diff
@@ -1,7 +1,39 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type AppProps = {};
+
+type AppState = {
+  pressedKey: string;
+};
+
+export class App extends React.Component<AppProps, AppState> {
+  state = {
+    pressedKey: '',
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', (e) => this.keyUpHandler(e));
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', this.keyUpHandler);
+  }
```

Comment:
why adding and removing the listener in a different manner?
it's still a callback here and there - `this.keyUpHandler` is perfectly fine
```suggestion
  componentDidMount() {
    document.addEventListener('keyup', this.keyUpHandler);
  }

  componentWillUnmount() {
    document.removeEventListener('keyup', this.keyUpHandler);
  }
```